repose-aggregator/src/docs/asciidoc/recipes/multi-tenant.adoc

```diff
@@ -0,0 +1,99 @@
+all of which need authorization applied to them.
+
+For the remainder of this recipe we are going to assume that you are trying to secure a resource this resource: `/a_tenant/do/something/on/another_tenant`.
+
+== Derive Tenants
+
+The first thing to do is to use the <<../filters/url-extractor-to-header.adoc#,URL Extractor to Header filter>> to pull out both tenants we care about into a common header.
+
+[source,xml]
+.url-extractor-to-header.cfg.xml
+----
+<url-extractor-to-header xmlns="http://docs.openrepose.org/repose/url-extractor-to-header/v1.0">
+    <extraction header="Requested-Tenant-Id" url-regex="/(\d*)/.*"> <!--1-->
+
+    <extraction header="Requested-Tenant-Id" url-regex=".*/do/something/on/(\d*)"/> <!--2-->
+</url-extractor-to-header>
+----
+<1> This regex will grab the first tenant with the assumption that the tenant is used throughout your api.
+<2> This regex is specifically grabbing the second tenant that we want to authorize on.
+
+== Configure Authentication
+
+For authentication we need to configure a couple of options on <<../filters/keystone-v2.adoc#,Keystone v2 filter>>.
+
+[source,xml]
+.keystone-v2.cfg.xml
+----
+<?xml version="1.0" encoding="UTF-8"?>
+<keystone-v2 xmlns="http://docs.openrepose.org/repose/keystone-v2/v1.0">
+    <identity-service uri="http://identity.example.com"
+        set-catalog-in-header="true"/> <!--1-->
+    <tenant-handling send-all-tenant-ids="true"/> <!--2-->
+</keystone-v2>
+----
+<1> If you want to do endpoint authorization you have to turn on this option.
+<2> This option ensures that all the tenants reported by keystone are added to the request for use in authorization.
+
+== Configure Tenant Authorization
+
+For basic tenant and endpoint authorization we use <<../filters/keystone-v2-authorization.adoc#,Keystone v2 Authorization filter>>.
+
+[source,xml]
+.keystone-v2-authorization.cfg.xml
+----
+<?xml version="1.0" encoding="UTF-8"?>
+<keystone-v2-authorization xmlns="http://docs.openrepose.org/repose/keystone-v2/v1.0">
+    <tenant-handling>
+        <validate-tenant>
+            <header-extraction-name>Requested-Tenant-Id</header-extraction-name> <!--1-->
+        </validate-tenant>
+    </tenant-handling>
+    <require-service-endpoint <!--2-->
+        public-url="https://service.example.com"
+        region="ORD"
+        name="OpenStackCompute"
+        type="compute"/>
+</keystone-v2-authorization>
+----
+<1> The header that contains the tenants to verify. This has to match the name provided in the Url Extractor Header filter.+
+    Every value found in this header has to be present among those returned by keystone.
+<2> The endpoint that you want to verify that the user has, note that this only works if you configure authentication to send the catalog.
+
+== Configure Role Based Access Control
+
+To get per tenant based access control you need to configure the <<../filters/api-validator.adoc#,API Validator filter>> nad the accompanying wadl.
```

Comment:
`nad` => `and`

Reply:
This should also probably link to `<<./role-based-access-control.adoc#3-utilize-tenanted-rax-roles,Role Based Access Control with multiple teanant>>` so the nuance about casing is brought to their attention.